bpf/lib/maps.h

```diff
@@ -99,6 +99,25 @@ struct bpf_elf_map __section_maps TUNNEL_MAP = {
 
 #endif
 
+#if defined (ENABLE_CUSTOM_CALLS) && defined (CUSTOM_CALLS_MAP)
+/* Private per-EP map for tail calls to user-defined programs.
+ * CUSTOM_CALLS_MAP is a per-EP map name, only defined for programs that need
+ * to use the map, so we do not want to compile this definition if
+ * CUSTOM_CALLS_MAP has not been #define-d.
+ */
+struct bpf_elf_map __section_maps CUSTOM_CALLS_MAP = {
+	.type		= BPF_MAP_TYPE_PROG_ARRAY,
+	.id		= CILIUM_MAP_CUSTOM_CALLS,
+	.size_key	= sizeof(__u32),
+	.size_value	= sizeof(__u32),
+	.pinning	= PIN_GLOBAL_NS,
+	.max_elem	= 2,	/* ingress and egress */
```

Comment:
The custom program seems to encode directionality in the `CB`, but this map hints at two different programs for ingress/egress. Which path do we want to support?

Reply:
Right, so there two ways to dissociate ingress/egress path:
- By attaching different programs in the two entries of the endpoint map.
- By encoding the direction and passing it down to the custom program.

The PR indeed supports both. I started with the different entries in the endpoint map, which has the advantage of skipping the tail call entirely for one direction if we only want to attach a program to the other direction. But I ended up attaching identical programs to collect data on both directions except the data would be stored in a different eBPF map entry based on the direction, and I figured that making the direction accessible to the custom program would be great to avoid compiling two otherwise identical programs.

With hindsight, I understand that mixing the two approaches may look confusing. I could remove the encoding of the direction in the `cb` buffer.